raft/src/main/java/org/apache/kafka/raft/internals/KRaftControlRecordStateMachine.java

```diff
@@ -279,19 +279,25 @@ private void handleBatch(Batch<?> batch, OptionalLong overrideOffset) {
             long currentOffset = overrideOffset.orElse(batch.baseOffset() + offsetDelta);
             switch (record.type()) {
                 case KRAFT_VOTERS:
+                    VoterSet voters = VoterSet.fromVotersRecord((VotersRecord) record.message());
+                    logger.info("Latest set of voters is {} at {}", voters, currentOffset);
                     synchronized (voterSetHistory) {
-                        voterSetHistory.addAt(currentOffset, VoterSet.fromVotersRecord((VotersRecord) record.message()));
+                        voterSetHistory.addAt(currentOffset, voters);
                     }
                     break;
 
                 case KRAFT_VERSION:
+                    KRaftVersion kraftVersion = KRaftVersion.fromFeatureLevel(
+                        ((KRaftVersionRecord) record.message()).kRaftVersion()
+                    );
+                    logger.info(
+                        "Latest {} is {} at {}",
```

Comment:
These will look like: `Lastest kraft.version is 1 at 100`. Can we add "offset" in there to make it clear what the last number is?

Reply:
Fixed.